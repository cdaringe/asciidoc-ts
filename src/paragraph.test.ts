import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("Paragraph - simple", async ({ expect }) => {
  const input = `A simple paragraph.`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Paragraph - with InlineElements", async ({ expect }) => {
  const input = `This is *foo* bar`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Paragraph - with newlines", async ({ expect }) => {
  const input = `Still\none\nparagraph.`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Paragraph - 2 newlines as 2 Paragraphs", async ({ expect }) => {
  const input = `Two\n\nparagraphs.`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
