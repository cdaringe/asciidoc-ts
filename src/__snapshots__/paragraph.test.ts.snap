// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Paragraph - 2 newlines as 2 Paragraphs 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "Two",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
    {
      "content": [
        {
          "content": "paragraphs.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;

exports[`Paragraph - simple 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "A simple paragraph.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;

exports[`Paragraph - with InlineElements 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "This is ",
          "type": "PlainText",
        },
        {
          "content": [
            {
              "content": "foo",
              "type": "PlainText",
            },
          ],
          "type": "ConstrainedBold",
        },
        {
          "content": "bar",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;

exports[`Paragraph - with newlines 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "Still",
          "type": "PlainText",
        },
        {
          "content": "one",
          "type": "PlainText",
        },
        {
          "content": "paragraph.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;
